components/usage/pkg/stripe/stripe.go

```diff
@@ -122,6 +130,38 @@ func (c *Client) updateUsageForCustomer(ctx context.Context, customer *stripe.Cu
 	}, nil
 }
 
+// GetUpcomingInvoice fetches the upcoming invoice for the given team
+func (c *Client) GetUpcomingInvoice(ctx context.Context, teamId string) (*StripeInvoice, error) {
+	query := fmt.Sprintf("metadata['teamId']:'%s'", teamId)
+	searchParams := &stripe.CustomerSearchParams{
+		SearchParams: stripe.SearchParams{
+			Query:   query,
+			Expand:  []*string{stripe.String("data.subscriptions")},
+			Context: ctx,
+		},
+	}
+	iter := c.sc.Customers.Search(searchParams)
+	if !iter.Next() {
+		return nil, fmt.Errorf("failed to find customer data for team %s", teamId)
+	}
+	customer := iter.Customer()
+	invoiceParams := &stripe.InvoiceParams{
+		Customer: stripe.String(customer.ID),
+	}
```

Comment:
What are the semantics when the search returns multiple customers? That should never happen, but we must be defensive here. I'd recommend collecting all the customers (or checking number of results), checking there's exactly one, and then returning.

Reply:
Good point. Copied from UpdateUsage without thinking about that further.